stan/math/rev/mat/fun/divide.hpp

```diff
@@ -9,18 +9,165 @@
 namespace stan {
 namespace math {
 
+namespace internal {
+template <int R, int C>
+class matrix_scalar_divide_vari_dv : public vari {
+ public:
+  int rows_;
+  int cols_;
+  vari* adjCRef_;
+  vari** adjResultRef_;
+  double invc;
+
+  explicit matrix_scalar_divide_vari_dv(const Eigen::Matrix<double, R, C>& m,
+                                        const var& c)
+      : vari(0.0),
+        rows_(m.rows()),
+        cols_(m.cols()),
+        adjCRef_(c.vi_),
+        adjResultRef_(ChainableStack::instance_->memalloc_.alloc_array<vari*>(
+            m.rows() * m.cols())) {
+    double c_d = c.val();
+    invc = 1.0 / c_d;
+    Eigen::Matrix<double, R, C> result = invc * m;
+    Eigen::Map<matrix_vi>(adjResultRef_, rows_, cols_)
+        = result.unaryExpr([](double x) { return new vari(x, false); });
+  }
+
+  virtual void chain() {
+    Eigen::Map<matrix_vi> adjResult(adjResultRef_, rows_, cols_);
+    adjCRef_->adj_
+        += -1.0 * invc
+           * (adjResult.adj().array() * adjResult.val().array()).sum();
+  }
+};
+
+template <int R, int C>
+class matrix_scalar_divide_vari_vd : public vari {
+ public:
+  int rows_;
+  int cols_;
+  vari** adjMRef_;
+  vari** adjResultRef_;
+  double invc;
```

Comment:
Same request here to use `invc_` and an initializer.

Reply:
Fixed underscore and moved invc_'s assignment to initializer.